lib/api/core/models/security/profile.js

```diff
@@ -21,23 +38,33 @@ function Profile () {
 Profile.prototype.isActionAllowed = function (requestObject, context, kuzzle) {
   var
     deferred = q.defer();
-
   if (this.roles === undefined || this.roles.length === 0) {
     return q(false);
   }
 
-  async.some(this.roles, (role, callback) => {
-    role.isActionAllowed(requestObject, context, kuzzle)
-      .then(isAllowed => {
-        callback(isAllowed);
+  this.getRoles(kuzzle)
+    .then(roles => {
+      async.some(roles, (role, callback) => {
+        role.isActionAllowed(requestObject, context, kuzzle)
```

Comment:
Should catch an error if, for some reason, `isActionAllowed` fails.

Easy way to do that with Q:

```js
role.isActionAllowed(requestObject, context, kuzzle).nodeify(callback);
```